src/pmp/algorithms/SurfaceRemeshing.cpp

```diff
@@ -181,6 +185,14 @@ void SurfaceRemeshing::preprocessing(std::string ear,
             min_edge_length_ *= 0.001;
             approx_error_ *= 0.001;
         }
+        if (unit == "m" && channel_left > 10)
+        {
+            channel_left *= 0.001;
+        }
+        if (unit == "m" && channel_right > 10)
+        {
+            channel_right *= 0.001;
+        } 
```

Comment:
I would suggest to remove this implicit processing and add to the documentation that channel_left and channel_right should be given in units of the input mesh